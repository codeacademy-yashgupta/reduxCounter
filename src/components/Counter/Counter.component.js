import React, {Component} from 'react';
import Button from './../Button/Button.component'

class Counter extends Component {
  incrementCount = () => {
    this.props.onAddClick();
  }
  decrementCount = () => {
    this.props.onsubtractClick();
  }

  render() {
    console.log(this.props)
    return (
      <div>
        <Button caption="Add" onClick ={this.incrementCount}></Button>
        <Button caption="Subtract" onClick={this.decrementCount}></Button>
      </div>
    );
  }
}
export default Counter;