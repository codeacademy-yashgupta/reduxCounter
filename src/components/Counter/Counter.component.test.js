import React from 'react';
import Counter from './Counter.component';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';


describe('Counter', () => {
  it('should render a Counter', () => {
    const tree = renderer.create(<Counter/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});