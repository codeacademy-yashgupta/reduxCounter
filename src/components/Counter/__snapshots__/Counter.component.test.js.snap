// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render a Counter 1`] = `
<div>
  <button
    className="button"
    onClick={[Function]}
  >
    Add
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Subtract
  </button>
</div>
`;
